provider/datagen/src/bin/datagen/mod.rs

```diff
@@ -114,12 +116,17 @@ fn main() -> eyre::Result<()> {
     };
     if config.without_fallback {
         let locale_families = match locales_intermediate {
+            AllLocales => eyre::bail!("--without-fallback needs an explicit locale list"),
             LanguageIdentifiers(lids) => lids,
-            LocaleFamilies(lfs) => eyre::bail!("--without-fallback needs an explicit locale list"),
+            LocaleFamilies(lfs) => lfs
+                .into_iter()
+                .map(|family| family.write_to_string().parse())
+                .collect::<Result<Vec<LanguageIdentifier>, icu_locid::ParserError>>()?,
```

Comment:
Note: This logic is a bit strange but I think it gets the job done. In Clap we parse the list to a `Vec<LocaleFamily>`, and then here I re-parse it as a `Vec<LanguageIdentifier>`. I guess another way to do this would be to parse to a `Vec<String>` in Clap and then pick the concrete type here.